Kernel/Process.cpp

```diff
@@ -1008,6 +925,18 @@ int Process::do_exec(NonnullRefPtr<FileDescription> main_program_description, Ve
         }
     }
 
+    if (interpreter_description) {
+        main_program_description->set_readable(true); // So that the dynamic loader can mmap the main program
+        auto main_program_fd = alloc_fd();
+        // FIXME: The proper way to do this is to pass a special auxilary vector to
+        // the dynamic loader. However, since the dynamic loader currently uses
+        // the normal libc _start, we can just pass data via the environ for now.
```

Comment:
Can we load both of them into memory at once? Main exe and loader? I think the BSDs and Linux pass that kind of info via the Auxiliary vector, but there's no reason we couldn't get hacky and pass a pointer to where the kernel loaded the text segment by pushing it on the stack, for example